### PR TITLE
Understanding category attribute from adapter

### PR DESCRIPTION
**Pull request by abhishkk**
Part 1 of [RFC](https://github.com/Microsoft/vstest-docs/pull/124):
Contains changes for understanding category attribute from adapter.

### CODE REVIEW

**Review by singhsarab — APPROVED** [review]
:shipit:

File: src/Microsoft.TestPlatform.Common/Utilities/AssemblyType.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Utilities
+{
+    public enum AssemblyType
+    {
+        NativeAndManaged,
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Common/Utilities/AssemblyType.cs]: do we need NativeAndManaged, can't we ask user specify Category attribute two time?

Comment (abhishkk) [reply src/Microsoft.TestPlatform.Common/Utilities/AssemblyType.cs]: Category attribute is not supported multiple times.
Adapter which supports both native and managed will not specify category attribute.

[the remark this answers is not in the document]
Comment (abhishkk) [reply src/Microsoft.TestPlatform.Common/Utilities/AssemblyType.cs]: done.

**Comment by abhishkk** [comment]
@dotnet-bot test this please

**Review by mayankbansal018 — APPROVED** [review]
:shipit:

**Comment by abhishkk** [comment]
@dotnet-bot test this

**Comment by abhishkk** [comment]
@dotnet-bot test this please

**Comment by abhishkk** [comment]
@dotnet-bot test this please

**Comment by abhishkk** [comment]
@dotnet-bot  test this please

**Pull request merged**